src/main/java/org/folio/circulation/services/ItemForTlrService.java

```diff
@@ -69,7 +70,27 @@ private Result<List<Item>> refusePageRequestWhenNoAvailablePageableItemsExist(Re
       return failedValidation(message, INSTANCE_ID, request.getInstanceId());
     }
 
-    return of(() -> availablePageableItems);
+    List<Item> finalAvailablePageableItems;
+    if (request.geItemLocationCode() != null) {
+      finalAvailablePageableItems = availablePageableItems.stream()
+        .filter(item -> request.geItemLocationCode().equals(item.getLocation().getCode()) ||
+            request.geItemLocationCode().equals(item.getLocation().getLibrary().getCode()) ||
+            request.geItemLocationCode().equals(item.getLocation().getCampus().getCode()) ||
+            request.geItemLocationCode().equals(item.getLocation().getInstitution().getCode())
+          )
+        .toList();
+      if (finalAvailablePageableItems.isEmpty()) {
+        String message = "Cannot create page TLR for this instance ID - no pageable available " +
+          "items found in forced location";
+        log.info("{}. Instance ID: {}, Forced location code {}",
```

Comment:
Where does the term "forced" come from? I don't see it mentioned in the Jira ticket. How about "requested"?